.github/actions/openvino_provider/action.yml

```diff
@@ -0,0 +1,156 @@
+name: 'OpenVINO provider'
+description: 'Provides OpenVINO build artifacts from a requested source'
+inputs:
+  revision:
+    description: 'Version of OpenVINO to provide - latest_available | HEAD | specific commit hash'
+    required: true
+  branch_name:
+    description: 'Branch of OpenVINO to take the revision from if no specific hash was provided. 
+                  Taken from github context by default'
+    required: false
+  ov_artifact_name:
+    description: "Name under which to upload provided OpenVINO build artifacts, set automatically by default"
+    required: false
+  commit_packages_to_provide:
+    description: "Comma-separated list of OpenVINO packages from post-commit to provide, all available by default"
+    required: false
+  commit_build_key:
+    description: 'public_linux_ubuntu_20_04_release | public_linux_ubuntu_22_04_release | 
+                  public_linux_ubuntu_24_04_release | public_windows_vs2019_release'
+    required: false
+  commit_share_path:
+    description: 'Path to share holding OpenVINO post-commit artifacts'
+    required: false
+    default: '/mount/build-artifacts'
+  nightly_build_key:
+    description: 'centos7_x86_64 | debian9_armhf | rhel8_x86_64 | ubuntu20_arm64 | ubuntu20_x86_64 | ubuntu22_x86_64 | 
+                  ubuntu24_x86_64 | macos_12_6_arm64 | macos_12_6_x86_64 | windows_x86_64'
+    required: false
+  nightly_package_source:
+    description: 'Source URL hosting OpenVINO nightly archives'
+    required: false
+    default: 'https://storage.openvinotoolkit.org/repositories/openvino/packages/nightly/latest.json'
+  nightly_pip_extra_url:
+    description: 'Source URL hosting OpenVINO nightly wheels, for passing to pip as --extra-index-url'
+    required: false
+    default: 'https://storage.openvinotoolkit.org/simple/wheels/nightly'
+outputs:
+  ov_artifact_name:
+    description: "Name of the artifact to upload OpenVINO build artifacts under"
+    value: ${{ steps.openvino_nightly_download.outputs.ov_artifact_name || steps.openvino_commit_output.outputs.ov_artifact_name }}
+  ov_package_path:
+    description: "Path to OpenVINO core package relative to the root of uploaded artifacts"
+    value: ${{ steps.openvino_nightly_download.outputs.ov_package_path || steps.openvino_commit_output.outputs.ov_package_path }}
+  ov_wheel_input:
```

Comment:
is it required to `cd ${{env.GITHUB_WORKSPACE}}/restored_artifact_from_commit` to install wheels ?
is it possible that ov_wheel_input will contain full path to find links to proper folder?

Reply:
The full path depends on the folder where the artifacts are downloaded in the build job, and we don't know it in advance (in case when provider and build are executed in different jobs, which is the main usage case for repos like tokenizers - see https://github.com/openvinotoolkit/openvino/pull/26050#discussion_r1716636537).

I'll check if it's possible to refer to some environment variable like INSTALL_DIR in find-links. If it does - the build job will need to set it before running pip install

Reply:
Added possibility to specify a full path to wheels via PIP_FIND_LINKS externally

Reply:
> The full path depends on the folder where the artifacts are downloaded in the build job, and we don't know it in advance (in case when provider and build are executed in different jobs, which is the main usage case for repos like tokenizers - see [#26050 (comment)](https://github.com/openvinotoolkit/openvino/pull/26050#discussion_r1716636537)).
> 
> I'll check if it's possible to refer to some environment variable like INSTALL_DIR in find-links. If it works - the build job will need to set it before running pip install, maybe it will be a bit more convenient

user will define it as action argument, like we do for download actions: https://github.com/mryzhov/openvino/blob/toml_build/.github/workflows/linux.yml#L661-L662